### PR TITLE
[kohaku] 6L/256d + Fourier PE + T_max=60 depth scaling

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

Wave 1 showed 4L → 5L (gilbert #76, no PE) improved from ~8.78% to ~7.47%. With Fourier PE, alphonse's 4L is at 7.21%. The question: does pushing depth to 6L with Fourier PE continue the improvement trend? The 2.7pp gap to AB-UPT is substantial; more depth may be needed to capture the full aerodynamic complexity. We double-check by testing 6L directly, skipping 5L (which alphonse is running) to cover the search space efficiently.

Reference: AB-UPT uses a deep transformer stack; we are likely capacity-limited at 4-5 layers for wall shear in particular.

**Target gap:** val_abupt from 7.21% → below 6.5% (stretch: below 6.0%).

## Instructions

Run exactly one training job:

```bash
cd target/ && python train.py \
  --model-num-layers 6 \
  --model-hidden-dim 256 \
  --no-use-ema \
  --lr 3e-4 \
  --lr-cosine-t-max 60 \
  --lr-min 1e-6 \
  --lr-warmup-epochs 5 \
  --fourier-pe \
  --no-compile-model \
  --nproc_per_node 4 \
  --kill-thresholds "500:train/loss<5,2000:val_primary/abupt_axis_mean_rel_l2_pct<25" \
  --wandb_group bengio-wave2
```

Key changes vs Wave 1 baseline:
- `--model-num-layers 6` (was 4) — two additional Transformer layers
- `--lr-cosine-t-max 60` (was 30) — even longer schedule for deeper model (empirical: deeper models need proportionally more epochs)
- `--lr-warmup-epochs 5` — important: longer warmup prevents early instability in deep models
- `--fourier-pe` — mandatory (Wave 1 lesson)
- `--no-use-ema` — same as baseline winner

Report val_primary metrics at the best checkpoint (lowest val_primary/abupt_axis_mean_rel_l2_pct).

## Baseline

Current best (PR #74, alphonse, Wave 1, W&B run `m9775k1v`):

| Metric | Current Best (val) | AB-UPT Target |
|--------|-------------------|--------------|
| `val_primary/abupt_axis_mean_rel_l2_pct` | **7.2091** | 4.51 |
| `val_primary/surface_pressure_rel_l2_pct` | 4.802 | 3.82 |
| `val_primary/wall_shear_rel_l2_pct` | 8.160 | 7.29 |
| `val_primary/volume_pressure_rel_l2_pct` | **4.166** ✓ | 6.08 |
| `val_primary/wall_shear_x_rel_l2_pct` | 7.109 | 5.35 |
| `val_primary/wall_shear_y_rel_l2_pct` | 9.100 | 3.65 |
| `val_primary/wall_shear_z_rel_l2_pct` | 10.869 | 3.63 |

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR CHECK-IN (2026-05-01 11:54 UTC)

**Stream status**: kohaku/128slices-fourier-pe is at step ~826K (epoch ~47/50). Current val metrics from W&B:
- abupt: **8.347%** (baseline: 7.209%)
- surf_p: 5.531%, vol_p: 5.893%
- wsy: 10.368%, wsz: 12.091%
- Trend slope: **+0.01269%/1k steps (WORSENING FAST — highest degradation rate)**

**Concern**: This is the fastest-deteriorating run. Using 128 point cloud slices (vs. standard) with Fourier PE is actively overfitting. The metric is worsening at 0.013%/1k steps — at this rate the final result will be ~8.5-9%+ at completion.

**Wave 2 FiLM sweep**: The kohaku-film-smallinit-sweep has film-zinit-B still running (~step 24K). Check those results carefully.

No action needed until the stream finishes — let it complete for the final record.

**Comment by morganmcg1** [comment]
**EP50 ASSESSMENT — DOES NOT BEAT BASELINE — STOP AND CLOSE** (2026-04-30 ~14:00 UTC)

Run `h7ve1hmb` (kohaku/6l-256d-fourier-pe-deeper) has reached step 890,850 (~ep50), finished at T_max=50 (configured as T_max=60 but pod appears to have stopped at ep50).

**Best checkpoint — step 570,143 (~ep32):**
| Metric | Best val | AB-UPT Target | vs Alphonse |
|--------|:---:|:---:|:---:|
| abupt_axis_mean_rel_l2_pct | **7.834%** | 4.51% | +0.62pp worse |
| surface_pressure_rel_l2_pct | 5.235% | 3.82% | +0.43pp worse |
| volume_pressure_rel_l2_pct | 5.569% | 6.08% | **BEATS TARGET** ✓ |
| wall_shear_x_rel_l2_pct | 7.480% | 5.35% | +0.37pp worse |
| wall_shear_y_rel_l2_pct | **9.465%** | 3.65% | +0.37pp worse |
| wall_shear_z_rel_l2_pct | 11.420% | 3.63% | +0.55pp worse |

**Gate decision: DOES NOT BEAT ALPHONSE BASELINE (7.209%). Closing.**

Analysis: 6L/256d depth scaling under T_max=60 peaks at ep32 (7.834%) vs 4L alphonse at ep30 (7.209%). The 8L alphonse run from Wave 2 ( at 11.334%) and this 6L result both confirm depth scaling in isolation does not help — the Wave 1 4L/256d is a sharp optimum. wsy=9.465% is slightly worse than alphonse 9.100%.

vol_p=5.569% beats AB-UPT target — this axis is now robustly beaten across 5+ runs.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #178 because EP50 ASSESSMENT: Run h7ve1hmb finished at step 890,850 (~ep50). Best checkpoint at step 570,143 (~ep32): val_abupt=7.834% — does not beat alphonse baseline of 7.209%. vol_p=5.569% beats AB-UPT target. wsy=9.465% is slightly worse than alphonse 9.100%. The 128-slice configuration adds compute without accuracy benefit. Closing as dead end. Thank you kohaku..

**Comment by morganmcg1** [comment]
STUDENT kohaku:

## Results

Acknowledging the advisor's EP50 assessment on run `h7ve1hmb` (kohaku/6l-256d-fourier-pe-deeper). The 6L/256d depth-scaling experiment with Fourier PE and T_max=60 ran to completion (~ep50) and **does not beat the alphonse 4L baseline**.

### Best checkpoint (step 570,143, ~ep32)

| Metric | Best val | Baseline (alphonse 4L) | Δ |
|--------|:---:|:---:|:---:|
| `val_primary/abupt_axis_mean_rel_l2_pct` | **7.834%** | 7.209% | +0.625pp ❌ |
| `val_primary/surface_pressure_rel_l2_pct` | 5.235% | 4.802% | +0.43pp ❌ |
| `val_primary/volume_pressure_rel_l2_pct` | 5.569% | 4.166% | +1.40pp ❌ (still beats AB-UPT 6.08% ✓) |
| `val_primary/wall_shear_y_rel_l2_pct` | 9.465% | 9.100% | +0.37pp ❌ |
| `val_primary/wall_shear_z_rel_l2_pct` | 11.420% | 10.869% | +0.55pp ❌ |

**Command used** (per PR instructions):
```bash
cd target/ && python train.py \
  --model-num-layers 6 --model-hidden-dim 256 --no-use-ema \
  --lr 3e-4 --lr-cosine-t-max 60 --lr-min 1e-6 --lr-warmup-epochs 5 \
  --fourier-pe --no-compile-model --nproc_per_node 4 \
  --kill-thresholds "500:train/loss<5,2000:val_primary/abupt_axis_mean_rel_l2_pct<25" \
  --wandb_group bengio-wave2
```

W&B run: `h7ve1hmb`

### What happened

Depth scaling 4L → 6L in isolation (with Fourier PE, T_max=60, warmup=5) does **not** continue the Wave 1 trend. The peak val_abupt of 7.834% at ep32 is worse than alphonse's 4L Fourier PE baseline (7.209% at ep30). Combined with the 8L alphonse run (Wave 2) at 11.334%, this confirms the advisor's hypothesis that **the 4L/256d Fourier PE configuration is a sharp optimum** on this dataset/budget, and depth scaling alone is not the right axis.

Volume pressure remains robustly under AB-UPT target (5.569% vs 6.08% target).

### Suggested follow-ups

- Wall shear axis decomposition (wsy=9.47%, wsz=11.42%) remains the dominant gap to AB-UPT target. Future runs should probably target axis-specific loss reweighting or rotation/symmetry-aware architectures rather than raw capacity scaling.
- If depth is to be revisited, pair it with width or token-budget changes (capacity-balanced sweeps) instead of depth-only.

Closing per advisor directive.
